### PR TITLE
tablecodec: improve package unit test code coverage to 85%

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

Improve test coverage

```
ok  	github.com/pingcap/tidb/tablecodec	0.016s	coverage: 85.8% of statements
```

### What is changed and how it works?

Add more test code

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @winkyao @jackysp

File: tablecodec/tablecodec_test.go [hunk 1]
```diff
@@ -64,31 +64,37 @@ func (s *testTableCodecSuite) TestRowCodec(c *C) {
...
+	row[4] = types.NewDatum(types.Set{Name: "a", Value: 0})
+	row[5] = types.NewDatum(types.BinaryLiteral{100})
 	// Encode
-	colIDs := make([]int64, 0, 3)
+	colIDs := make([]int64, 0, len(row))
 	for _, col := range cols {
 		colIDs = append(colIDs, col.id)
 	}
-	sc := &stmtctx.StatementContext{TimeZone: time.UTC}
+	sc := &stmtctx.StatementContext{TimeZone: time.Local}
```
Comment (jackysp) [inline tablecodec/tablecodec_test.go]: why was this changed?

Comment (tiancaiamao) [reply tablecodec/tablecodec_test.go]: Cover here https://github.com/pingcap/tidb/blob/master/tablecodec/tablecodec.go#L255

**Comment by jackysp** [comment]
Please fix CI.
